torch/ao/quantization/fx/_model_report/model_report_visualizer.py

```diff
@@ -435,17 +445,94 @@ def generate_table_visualization(self, feature_filter: str = "", module_fqn_filt
 
         print(table_str)
 
+    def _get_plotting_data(self, table_dict: Dict[str, Tuple[List, List]]) -> Tuple[bool, List, List]:
+        r"""
+        Takes in the table information as a dictionary and outputs both the x and y data
+        required to do the plotting.
+
+        Args:
+            table_dict ( Dict[str, Tuple[List, List]]): maps each of the:
+                TENSOR_LEVEL_INFO -> tensor_headers, tensor_table
+                CHANNEL_LEVEL_INFO -> channel_headers, channel_table
+
+        Returns a tuple of three elements:
+            If the data is per_channel or not
+            A list of the x values to plot
+            A list of the y values to plot
+        """
+        tensor_headers, tensor_table = table_dict[self.TABLE_TENSOR_KEY]
+        channel_headers, channel_table = table_dict[self.TABLE_CHANNEL_KEY]
+
+        # make sure it is only 1 feature that is being plotted
+        # get the number of features in each of these
+        tensor_info_features_count = len(tensor_headers) - ModelReportVisualizer.NUM_NON_FEATURE_TENSOR_HEADERS
+        channel_info_features_count = len(channel_headers) - ModelReportVisualizer.NUM_NON_FEATURE_CHANNEL_HEADERS
+
+        # keep track of per_channel or not
+        data_is_per_channel: bool = False
+
+        x_data: List = []
+        y_data: List[List] = []
+        # the feature will either be a tensor feature or channel feature
+        if tensor_info_features_count == 1:
```

Comment:
branch condition looks weird, if it can only be one of them, can we have a variable for that, e.g. `is_show_tensor_info` and branch on that

Reply:
it's possible that the user tries to have multiple features included at once for plotting, which is not something that we want and we want to alert them of, which is why we have the explicit looks and checks. I updated it a bit so that it looks less hardcoded.